Updated job requirements, added DESIRED_Sites/UNDESIRED_Sites. (#163)
OSGConnect/connectbook: start/resources/requirements.md

either focus on, or avoid, certain execution sites.

# Requirements

The `requirements` attribute is formatted as an expression, so you can use logical
operators to combine multiple requirements where `&&` is used for AND and
`||` used for OR. For example, the following `requirements` statement will direct
jobs only to 64 bit RHEL (Red Hat Enterprise Linux) 8 nodes.

    requirements = OSGVO_OS_STRING == "RHEL 8" && Arch == "X86_64"

Alternatively, if you have code which can run on either RHEL 7 or 8, you can use OR:

    requirements = (OSGVO_OS_STRING == "RHEL 7" || OSGVO_OS_STRING == "RHEL 8") && Arch == "X86_64"

Note that parentheses placement is important for controling how the logical operations
are interpreted by HTCondor.
 
Another common requirement is to land on a node which has CVMFS.
Then the `requirements` would be:

	requirements = HAS_oasis_opensciencegrid_org == True

## Additional Feature-Specific Attributes

There are many attributes that you can use with `requirements`. To see what values
you can specify for a given attribute you can run the following command while
connected to your login node:

> 	$ condor_status -pool cm-1.ospool.osg-htc.org -af {ATTR_NAME} | sort -u

Reviewer: do we even need the pool options? 
```suggestion
	$ condor_status -af {ATTR_NAME} | sort -u
```

	
For example, to see what values you can specify for the OSGVO_OS_STRING attribute run:
	

> 	$ condor_status -pool cm-1.ospool.osg-htc.org -af OSGVO_OS_STRING | sort -u

Reviewer: ```suggestion
	$ condor_status -af OSGVO_OS_STRING | sort -u
```

    RHEL 7
    RHEL 8

This means that we can specify an OS version of `RHEL 7` or `RHEL 8`. Alternatively

> you will find many attrubites will take the boolean values `true` or `false`.

Reviewer: ```suggestion
you will find many attributes will take the boolean values `true` or `false`.
```

- **OSGVO_OS_VERSION** - Version of the operating system

- **OSGVO_OS_STRING** - Combined OS name and version. Common values are
  _RHEL 7_ and _RHEL 8_. Please see the requirements string above on the
  recommended setup.

- **OSGVO_CPU_MODEL** - The CPU model identifier string as presented in
  /proc/cpuinfo

- **HAS_CVMFS_oasis_opensciencegrid_org** - Attribute specifying
  the need to access specific oasis /cvmfs file system repositories.

> - **CUDACapability** - For GPU jobs, specifies the CUDA compute capability.

Reviewer: ```suggestion
- **CUDACapability** - For GPU jobs, specifies the CUDA compute capability.
See our [GPU guide](5000653025) for more details. 
```

Author: Added this as a commit as github would not let me direct commit it.

# Specifying Sites / Avoiding Sites

To run your jobs on a list of specific execution sites, or avoid a set of 
sites, use the `+DESIRED_Sites`/`+UNDESIRED_Sites` attributes in your job

> submit file. These attributes should only be used as a last resort. For

Reviewer: ```suggestion
submit file. **These attributes should only be used as a last resort.** For
```

example, it is much better to use feature attributes (see above) to make
your job go to nodes matching what you really require, than to broadly

> allow/block whole sites.

Reviewer: ```suggestion
allow/block whole sites. We encourage you to contact the facilitation team before taking this action, to make sure it is right for you. 
```

To avoid certain sites, first find the site names. You can find a 

> current list by quering the pool:

Reviewer: ```suggestion
current list by querying the pool:
```

    condor_status -af GLIDEIN_Site | sort -u

> In your submit file, add a comma separated list of those site like:

Reviewer: ```suggestion
In your submit file, add a comma separated list of sites like:
```

  /proc/cpuinfo

- **HAS_CVMFS_oasis_opensciencegrid_org** - Attribute specifying
  the need to access specific oasis /cvmfs file system repositories.

- **CUDACapability** - For GPU jobs, specifies the CUDA compute capability.
  See our [GPU guide](5000653025) for more details.

# Specifying Sites / Avoiding Sites

To run your jobs on a list of specific execution sites, or avoid a set of 
sites, use the `+DESIRED_Sites`/`+UNDESIRED_Sites` attributes in your job
submit file. **These attributes should only be used as a last resort.** For
example, it is much better to use feature attributes (see above) to make
your job go to nodes matching what you really require, than to broadly
allow/block whole sites. We encourage you to contact the facilitation team before taking this action, to make sure it is right for you. 

To avoid certain sites, first find the site names. You can find a 
current list by querying the pool:

    condor_status -af GLIDEIN_Site | sort -u

In your submit file, add a comma separated list of sites like:

    +UNDESIRED_Sites = "ISI,SU-ITS"

Those sites will now be exluded from the set of sites your job can

> run at.

Reviewer: also say something about `+DesiredSites` or omit it above.

Author: Addressed in new commit
